client/src/pages/lessons/edit.tsx

```diff
@@ -396,692 +360,606 @@ const LessonEdit = (props: {
                   fullWidth
-                  InputLabelProps={{
-                    shrink: true,
-                  }}
-                  value={lessonUnderEdit.lesson.media.url || ""}
-                  onChange={(e) => {
-                    setLesson(
-                      {
-                        ...(lessonUnderEdit.lesson || newLesson),
-                        media: {
-                          ...(lessonUnderEdit.lesson || newLesson).media,
-                          type: MediaType.IMAGE,
-                          url: (e.target.value as string) || "",
-                        },
-                      },
-                      true
-                    );
-                  }}
-                  variant="outlined"
-                />
-                <img
-                  className={classes.thumbnail}
-                  data-cy="image-thumbnail"
-                  src={lessonUnderEdit.lesson.media.url}
-                  onClick={() => {
-                    window.open(
-                      lessonUnderEdit.lesson?.media?.url || "",
-                      "_blank"
-                    );
-                  }}
-                />
-              </div>
-            </Grid>
-          ) : (
-            <></>
-          )}
-          {lessonUnderEdit.lesson.media &&
-          lessonUnderEdit.lesson.media.type === MediaType.VIDEO ? (
-            <>
-              <Grid item xs={12}>
-                <TextField
-                  data-cy="video-url"
-                  label="Video"
-                  placeholder="YouTube Video URL"
-                  required
                   multiline
-                  maxRows={4}
-                  fullWidth
                   InputLabelProps={{
                     shrink: true,
                   }}
-                  value={lessonUnderEdit.lesson.media.url || ""}
-                  onChange={(e) => {
-                    setLesson(
-                      {
-                        ...(lessonUnderEdit.lesson || newLesson),
-                        media: {
-                          ...(lessonUnderEdit.lesson || newLesson).media,
-                          type: MediaType.VIDEO,
-                          url: (e.target.value as string) || "",
-                        },
-                      },
-                      true
-                    );
-                  }}
+                  value={lessonUnderEdit.lesson?.name || ""}
+                  onChange={handleLessonName}
                   variant="outlined"
                 />
               </Grid>
-              <Grid item xs={12}>
-                <Grid
-                  container
-                  // TODO: With MUI v5 We can dimply do: rowSpacing={2}
-                  direction={"row"}
-                  spacing={2}
-                >
-                  <Grid item xs={6}>
+              <Grid item xs={4}>
+                <FormControl className={classes.selectForm} variant="outlined">
```

Comment:
For example, maybe move this "lesson format selector" into its own component file